content/en/guide/v10/web-components.md

```diff
@@ -58,3 +58,65 @@ Preact normalizes the casing of standard built-in DOM Events, which is how we ca
 // Add handler for "ionChange" event (note the casing)
 <my-foo onionChange={() => console.log('ionChange')} />
 ```
+
+## Creating a Web Component
+
+Any Preact component can be turned into a web component with [preact-custom-element](https://github.com/preactjs/preact-custom-element), a very thin wrapper that adheres to the Custom Elements v1 spec.
+
+```jsx
+import register from 'preact-custom-element';
+
+const Greeting = ({ name = 'World' }) => (
+  <p>Hello, {name}!</p>
+);
+
+register(Greeting, 'x-greeting', ['name']);
+//          ^            ^           ^
+//          |      HTML tag name     |
+//   Component definition      Observed attributes
+```
+
+> Note: As per the [Custom Element Specification](http://w3c.github.io/webcomponents/spec/custom/#prod-potentialcustomelementname), the tag name must contain a hyphen (`-`).
+
+Use the new tag name in HTML, attribute keys and values will be passed in as props:
+
+```html
+<x-greeting name="Billy Jo"></x-greeting>
+```
+
+Output:
+
+```html
+<p>Hello, Billy Jo!</p>
+```
+
+### Observed Attributes
+
+In contrast to Preact components, Web Components require explicitly stating the names of attributes you want to observe. Otherwise the component will not receive any attribute changes. These can be specified via the third parameter that's passed to the `register()` function:
+
+```jsx
+// Listen to changes to the `name` attribute
+register(Greeting, 'x-greeting', ['name']);
+```
+
+If you omit the third parameter to `register()`, the list of attributes to observe can be specified using a static `observedAttributes` property on your Component. This also works for the Custom Element's name, which can be specified using a `tagName` static property:
+
+If the third parameter is not left out we'll try to infer
```

Comment:
Would you tell me that `infer` mean in this sentence concretely?

```
If the third parameter is not left out, we'll try to infer.
```

Reply:
Yeah I kinda stopped mid sentence there 🙈 👀